src/util/tempdir.cpp

```diff
@@ -64,9 +66,11 @@ std::string get_temporary_directory(const std::string &name_template)
       prefixed_name_template+='/';
     prefixed_name_template+=name_template;
 
-    char t[1000];
-    strncpy(t, prefixed_name_template.c_str(), 1000);
-    const char *td = mkdtemp(t);
+    std::vector<char> t;
+    // add one byte for the zero
+    t.resize(prefixed_name_template.size()+1);
+    memcpy(t.data(), prefixed_name_template.c_str(), t.size());
```

Comment:
Is this guaranteed to be 0-terminated? I'd do:
````c++
std::vector<char> t(prefixed_name_template.begin(), prefixed_name_template.end());
t.push_back('\0');
````
(and this entire code will become unnecessary with C++17).

Reply:
ok